tools/override-generator/main.go

```diff
@@ -131,13 +131,17 @@ func run(a mainArgs, stderr io.Writer) error {
 	})
 
 	var outputOverrides []*rule.Rule
+	defaultAttrs := map[string]string{
+		_buildFileGenerationAttr: a.defaultBuildFileGeneration,
+		_buildFileProtoModeAttr:  a.defaultBuildFileProtoMode,
+	}
 
 	// Iterate over all repositories and convert them to override rules
 	// The repos are ordered by "name", and the sets are sorted, so the output
 	// will be deterministic.
 	for _, r := range repos {
 		if r.Kind() == "go_repository" {
-			repoOverrides := goRepositoryToOverrideSet(r)
+			repoOverrides := goRepositoryToOverrideSet(r, defaultAttrs)
```

Comment:
Actually need to refactor this a little. Since even though we don't want the override when it's just default attributes. We will need it if there's a `gazelle-override` created, since if there is, it falls back to `auto`